master/internal/db/postgres.go

```diff
@@ -1447,6 +1485,114 @@ func (db *PgDB) UpdateStep(
 	if err != nil {
 		return errors.Wrapf(err, "error finding step (%v, %v) to update", trialID, totalBatches)
 	}
+	return db.updateStep(step, newState, metrics)
+}
+
+// AddTrainingMetrics adds a completed step to the database with the given training metrics.
```

Comment:
Can you briefly explain the idea here? why do you need to migrate steps on the fly rather than in migration scipt?

Reply:
There is no migration here, which part are you referring to?

Reply:
sry. I might misunderstand the logic here. Because in order to add a training metrics, you roll back all the training metrics and validations. At least the validations seem very irrelevant. Why do you need to add interdependency here?

Reply:
talk offline, gonna change all this and store multiple runs